File: samples/bot-auth0-adaptivecard/nodejs/services/TokenStore.js
class TokenStore {
    constructor() {
        this.tokens = new Map();
    }

    setToken(userId, token) {
        this.tokens.set(userId, token);
    }

    getToken(userId) {
        return this.tokens.get(userId) || null;
    }

    removeToken(userId) {
        this.tokens.delete(userId);
    }

    static getInstance() {
        if (!TokenStore.instance) {
            TokenStore.instance = new TokenStore();
        }
        return TokenStore.instance;
    }
}

module.exports = TokenStore.getInstance();

File: samples/bot-auth0-adaptivecard/python/services/token_store.py
from typing import Optional
import threading

class TokenStore:
    _instance = None
    _lock = threading.Lock()

    def __new__(cls):
        if cls._instance is None:
            with cls._lock:
                if cls._instance is None:
                    cls._instance = super(TokenStore, cls).__new__(cls)
                    cls._instance._tokens = {}
        return cls._instance

    def set_token(self, user_id: str, token: str):
        self._tokens[user_id] = token

    def get_token(self, user_id: str) -> Optional[str]:
        return self._tokens.get(user_id)

    def remove_token(self, user_id: str):
        self._tokens.pop(user_id, None)

    def clear_tokens(self):
        self._tokens.clear()